test/gpu/native/studies/transpose/transpose.chpl

```diff
@@ -3,15 +3,59 @@ use GpuDiagnostics;
 use GPU;
 use Time;
 
+enum implType {
+    naive, clever, lowlevel
+}
+use implType;
+
 config const perftest = false;
 config const numTrials = 10;
-config const useNaive = false;
+config const impl = naive;
 config const sizeX = 2048*8,
              sizeY = 2048*8;
 config param blockSize = 16;
 config param blockPadding = 1;
 config type dataType = real(32);
 
+inline proc transposeNaive(original, output) {
+  foreach (x,y) in original.domain do output[y,x] = original[x,y];
```

Comment:
Just to check my understanding:

This will lower into a pair of nested loops and we only do the kernel launch for the outer loop right?

Also (assuming we do expect this to run on the GPU) can you add `assertOnGpu()` here?

Reply:
> This will lower into a pair of nested loops and we only do the kernel launch for the outer loop right?

Yes, the outer loop will be GPUized and the inner loop will be inside the GPU kernel.

> Also (assuming we do expect this to run on the GPU) can you add assertOnGpu() here?

Sure thing!

Reply:
> Yes, the outer loop will be GPUized and the inner loop will be inside the GPU kernel.

Ok makes sense. Could you add a comment explaining that? Normally I wouldn't care but I could see pointing users to this test as an example of Chapel GPU code so I think it'd good to add one.